scripts/migrations/8-fraction.py

```diff
@@ -0,0 +1,29 @@
+"""
+This migration implements spec change https://github.com/nearprotocol/NEPs/pull/58.
+
+Changes:
+ - Change `gas_price_adjustment_rate`, `protocol_reward_percentage`, `developer_reward_percentage`,
+ and `max_inflation_rate` to fractions.
+"""
+
+
+import sys
+import os
+import json
+from collections import OrderedDict
+
+home = sys.argv[1]
+output_home = sys.argv[2]
+
+config = json.load(open(os.path.join(home, 'output.json')), object_pairs_hook=OrderedDict)
+
+assert config['protocol_version'] == 7
+
+config['protocol_version'] = 8
+config['gas_price_adjustment_rate'] = [1, 100]
+config['protocol_reward_percentage'] = [1, 10]
+config['developer_reward_percentage'] = [3, 10]
+config['max_inflation_rate'] = [1, 20]
+config['runtime_config']['transaction_costs']['burnt_gas_reward'] = [3, 10]
```

Comment:
It is fine now, but I want to rise some awareness, that we need to avoid this overriding style in the future, right? (we don't want to override any existing settings with just default values)

Also, should we remove `protocol_reward_percentage`?

Reply:
What do you suggest instead of overriding?
> Also, should we remove protocol_reward_percentage?
Why?

Reply:
> What do you suggest instead of overriding?

I suggest computing from the previous value, so if that was percentage integer, it will become `[old value, 100]`

> Why?

My bad, nevermind.

Reply:
The problem is that the internal representation is an irreducible fraction so it is impossible to have 100 as the denominator if the numerator is not relatively prime to 100.

Reply:
Python has built-in `math.gcd`...

Reply:
> Python has built-in math.gcd

Not sure what you mean here. It is not about manually reducing the fraction. The representation has to conform to internal representation otherwise the check on genesis will fail.

Reply:
It seems that I don't understand what internal representation you refer to. Given it is part of the genesis config, and it should be updated with great caution by someone who understands what they are going, I am fine to just let it go (though I admit, I don't understand why the existing value, e.g. 5%, we cannot convert it to a fraction representation, i.e. 1/20, and instead we override it to 3/10)

Reply:
We didn't override it to 3/10. We changed it 1/20.